depthcharge/data/hdf5.py

```diff
@@ -389,29 +391,24 @@ def _get_parser(self, ms_data_file):
 
         raise ValueError("Only MGF files are supported.")
 
-    def _assemble_metadata(self, parser):
-        """Assemble the metadata.
+    def get_spectrum(self, file_index, spectrum_index):
+        """Access a mass spectrum.
 
         Parameters
         ----------
-        parser : MzmlParser or MgfParser
-            The parser to use.
+        file_index : int
+            The group index of the MS data file.
+        spectrum_index : int
+            The index of the mass spectrum within the file.
 
         Returns
         -------
-        numpy.ndarray of shape (n_spectra,)
-            The file metadata.
+        tuple of numpy.ndarray
+            The m/z values, intensity values, precurosr m/z, precurosr charge,
+            and the spectrum annotation.
         """
-        meta_types = [
-            ("precursor_mz", np.float32),
-            ("precursor_charge", np.uint8),
-            ("offset", np.uint64),
-            ("annotations", h5py.string_dtype()),
-        ]
-
-        metadata = np.empty(parser.n_spectra, dtype=meta_types)
-        metadata["precursor_mz"] = parser.precursor_mz
-        metadata["precursor_charge"] = parser.precursor_charge
-        metadata["offset"] = parser.offset
-        metadata["annotations"] = parser.annotations
-        return metadata
+        spec_info = super().get_spectrum(file_index, spectrum_index)
+        grp = self._handle[str(file_index)]
+        annotations = grp["annotations"]
+        spec_ann = annotations[spectrum_index].decode()
+        return *spec_info, spec_ann
```

Comment:
Syntax error in the current form. Changed to `return *spec_info, spec_ann` to make fix.

Reply:
🤦‍♂️ Thanks for catching that.

Reply:
I actually don't get an error... nor did the GitHub Actions tests.

Reply:
Ah - this is only valid in Python 3.8+.